tmp.proto

```
@@ -0,0 +1,68 @@
+// TODO "orderId" you would think would be unambiguous.
+
+message AppState {  // "App" is just generic for YNAB
+  number schemaVersion;
+  number serverKnowledge;
+}
+
+// The initial state is UNKNOWN. The final states are NOT_NEEDED, SUCCEEDED, and FAILED.
+// UNKNOWN can transition to NOT_NEEDED or NEEDS_SYNC.
+// NEEDS_SYNC and NEEDS_RETRY can transition to SUCCEEDED, NEEDS_RETRY or FAILED.
+enum AppTransactionSyncState {
+  UNKNOWN;
```

Buf (1.47.2)

syntax error: unexpected ';', expecting '='

(COMPILE)